### PR TITLE
Remove ifdefs from winforms

### LINKED ISSUE #39: Remove unused managed defines from S.W.F code
There are a few pre-processor directives in S.W.F code which might not be required.
Scrub the code and keep only those which are required.

### PR DESCRIPTION
**Pull request by hughbe**
## Proposed Changes
- Remove never defined ifdefs from winforms
- Remove always defined ifdefs from winforms

The remaining list is
- DEBUG
- ~DEBUG_LAYOUT~
- ~DEBUG_PAINT~
- ~DEBUG_PAINT_ANCHOR~
- ~DEBUG_UPDOWN~
- ~FINALIZATION_WATCH~
- ~GDI_FINALIZATION_WATCH~
- ~LAYOUT_PERFWATCH~
- ~PBRS_PAINT_DEBUG~
- ~TRACK_HDC~

What should we do with these? I didn't want to delete as they may be useful, but all the above except `DEBUG` (which should stay, obviously), are never used define. Should we delete this code?

Fixes #39

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/3169)

### CODE REVIEW

**Comment by RussKie** [comment]
@Tanya-Solyanik any objections?

**Comment by hughbe** [comment]
Also would appreciate some suggestions on the remaining ifdefs listed above

**Comment by JeremyKuhne** [comment]
Keep `#ifdef DEBUG`. We should create an issue to consider how we want to track finalization so that we can shake out issues in disposal as we're clearly having problems in our test runs (to replace the two FINALIZATION_WATCH defines).

File: src/System.Windows.Forms/src/misc/GDI/DeviceContext.cs [hunk 1]
```diff
@@ -2,7 +2,6 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-// #define TRACK_HDC
 // #define GDI_FINALIZATION_WATCH
```
Comment (RussKie) [inline src/System.Windows.Forms/src/misc/GDI/DeviceContext.cs:5]: Looks like this was missed

Comment (weltkante) [reply src/System.Windows.Forms/src/misc/GDI/DeviceContext.cs:5]: I think FINALIZATION_WATCH is now back in [as requested](https://github.com/dotnet/winforms/pull/3169#issuecomment-632419583)

> Delete remaining except FINALIZATION-WATCH

Comment (RussKie) [reply src/System.Windows.Forms/src/misc/GDI/DeviceContext.cs:5]: It is being removed in other places though.
I created https://github.com/dotnet/winforms/issues/3348 to track/reinstate finalization watch.

**Comment by hughbe** [comment]
Updated

File: src/System.Windows.Forms/src/misc/GDI/DeviceContext.cs (hunk 1 above)
Comment (weltkante) [reply src/System.Windows.Forms/src/misc/GDI/DeviceContext.cs]: Oh, didn't notice it was inconsistent, probably should be updated to either remove all of them or keep all of them

Comment (hughbe) [reply src/System.Windows.Forms/src/misc/GDI/DeviceContext.cs]: Thats what I did :)

**Comment by RussKie** [comment]
Looks like tests in x64 just crash without any additional info... 
```
The command ""F:\workspace\_work\1\s\.dotnet\dotnet.exe" exec --depsfile "F:\workspace\_work\1\s\artifacts\bin\System.Windows.Forms.Tests\Debug\netcoreapp5.0\System.Windows.Forms.Tests.deps.json" --runtimeconfig "F:\workspace\_work\1\s\artifacts\bin\System.Windows.Forms.Tests\Debug\netcoreapp5.0\System.Windows.Forms.Tests.runtimeconfig.json"  "F:\workspace\_work\1\s\.packages\xunit.runner.console/2.4.1/tools/netcoreapp2.0/xunit.console.dll" "F:\workspace\_work\1\s\artifacts\bin\System.Windows.Forms.Tests\Debug\netcoreapp5.0\System.Windows.Forms.Tests.dll" -noautoreporters -xml "F:\workspace\_work\1\s\artifacts\TestResults\Debug\System.Windows.Forms.Tests_netcoreapp5.0_x64.xml" -html "F:\workspace\_work\1\s\artifacts\TestResults\Debug\System.Windows.Forms.Tests_netcoreapp5.0_x64.html"  > "F:\workspace\_work\1\s\artifacts\log\Debug\System.Windows.Forms.Tests_netcoreapp5.0_x64.log" 2>&1" exited with code -1073740940.
```

**Comment by RussKie** [comment]
...and it passed 😕

**Pull request merged**